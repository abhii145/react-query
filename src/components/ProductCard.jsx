import { Link } from "react-router"
import PropTypes from "prop-types"

const ProductCard = ({ products }) => {
  return (
    <div className="bg-white">
      <div className="mx-auto max-w-2xl px-4 py-16 sm:px-6 sm:py-24 lg:max-w-7xl lg:px-8">
        <div className="mt-6 grid grid-cols-1 gap-x-6 gap-y-10 sm:grid-cols-2 lg:grid-cols-4 xl:gap-x-8">
          {products?.map((product) => {
            return (
              <div key={product.id} className="group relative">
                <img
                  loading="lazy"
                  alt={product.title}
                  src={product.thumbnail}
                  className="aspect-square w-full rounded-md bg-gray-200 object-contain group-hover:opacity-75 lg:aspect-auto lg:h-80"
                />
                <div className="mt-4 flex justify-between">
                  <div>
                    <h3 className="text-sm text-gray-700">
                      <Link to={`/product/${product.id}`}>
                        <span aria-hidden="true" className="absolute inset-0" />
                        {product.title}
                      </Link>
                    </h3>
                    <p className="mt-1 text-sm text-gray-500">
                      {product.category}
                    </p>
                  </div>
                  <p className="text-sm font-medium text-gray-900">
                    â‚¹ {product.price}
                  </p>
                </div>
              </div>
            )
          })}
        </div>
      </div>
    </div>
  )
}
ProductCard.propTypes = {
  products: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      thumbnail: PropTypes.string.isRequired,
      category: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired,
    })
  ).isRequired,
}

export default ProductCard
