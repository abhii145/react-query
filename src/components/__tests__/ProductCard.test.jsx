import { render, screen } from "@testing-library/react"
import { BrowserRouter as Router } from "react-router-dom"
import ProductCard from "../ProductCard"
import { it, expect, describe } from "vitest"

const products = [
  {
    id: 1,
    title: "Product 1",
    thumbnail: "image1.jpg",
    category: "Category 1",
    price: 100,
  },
  {
    id: 2,
    title: "Product 2",
    thumbnail: "image2.jpg",
    category: "Category 2",
    price: 200,
  },
]

describe("ProductCard", () => {
  it("renders product cards", () => {
    render(
      <Router>
        <ProductCard products={products} />
      </Router>
    )
    expect(screen.getByText("Product 1")).toBeInTheDocument()
    expect(screen.getByText("Product 2")).toBeInTheDocument()
  })

  it("renders product details", () => {
    render(
      <Router>
        <ProductCard products={products} />
      </Router>
    )
    expect(screen.getByText("Category 1")).toBeInTheDocument()
    expect(screen.getByText("100")).toBeInTheDocument()
  })
})
