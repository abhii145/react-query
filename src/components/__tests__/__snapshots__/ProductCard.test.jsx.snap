// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductCard > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-white"
  >
    <div
      class="mx-auto max-w-2xl px-4 py-16 sm:px-6 sm:py-24 lg:max-w-7xl lg:px-8"
    >
      <div
        class="mt-6 grid grid-cols-1 gap-x-6 gap-y-10 sm:grid-cols-2 lg:grid-cols-4 xl:gap-x-8"
      >
        <div
          class="group relative"
        >
          <img
            alt="Product 1"
            class="aspect-square w-full rounded-md bg-gray-200 object-contain group-hover:opacity-75 lg:aspect-auto lg:h-80"
            loading="lazy"
            src="image1.jpg"
          />
          <div
            class="mt-4 flex justify-between"
          >
            <div>
              <h3
                class="text-sm text-gray-700"
              >
                <a
                  data-discover="true"
                  href="/product/1"
                >
                  <span
                    aria-hidden="true"
                    class="absolute inset-0"
                  />
                  Product 1
                </a>
              </h3>
              <p
                class="mt-1 text-sm text-gray-500"
              >
                Category 1
              </p>
            </div>
            <p
              class="text-sm font-medium text-gray-900"
            >
              100
            </p>
          </div>
        </div>
        <div
          class="group relative"
        >
          <img
            alt="Product 2"
            class="aspect-square w-full rounded-md bg-gray-200 object-contain group-hover:opacity-75 lg:aspect-auto lg:h-80"
            loading="lazy"
            src="image2.jpg"
          />
          <div
            class="mt-4 flex justify-between"
          >
            <div>
              <h3
                class="text-sm text-gray-700"
              >
                <a
                  data-discover="true"
                  href="/product/2"
                >
                  <span
                    aria-hidden="true"
                    class="absolute inset-0"
                  />
                  Product 2
                </a>
              </h3>
              <p
                class="mt-1 text-sm text-gray-500"
              >
                Category 2
              </p>
            </div>
            <p
              class="text-sm font-medium text-gray-900"
            >
              200
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
